src/components/wizard-internet/wizard-internet.js

```diff
@@ -8,17 +8,128 @@ export default module('app.components.wizard-internet', [])
     },
     // TODO: handle vpn
     controller: class WizardContactCtrl {
+
       constructor($scope) {
         'ngInject';
 
         this.newInternet = {};
+        this.internetTunnelEnabled = false;
+        this.meshTunnelEnabled = false;
+        this.internetTunnelConf = [
+          {
+            fileExtensions: '.ovpn,.cnf,text/*',
+            property: 'config',
+            tunnelType: 'internettunnel',
+            required: true,
+          },
+          {
+            fileExtensions: '.crt,text/*',
+            tunnelType: 'internettunnel',
+            property: 'cert',
+          },
+          {
+            fileExtensions: '.crt,text/*',
+            tunnelType: 'internettunnel',
+            property: 'cacert',
+          },
+          {
+            fileExtensions: '.key,text/*',
+            tunnelType: 'internettunnel',
+            property: 'key',
+          },
+          {
+            fileExtensions: '.key,text/*',
+            tunnelType: 'internettunnel',
+            property: 'takey',
+          },
+        ];
+
+        this.internetTunnelConfSecret = [
+          {
+            fileExtensions: '*',
+            tunnelType: 'internettunnel',
+            property: 'auth-user-pass',
+          },
+          {
+            fileExtensions: '*',
+            tunnelType: 'internettunnel',
+            property: 'secret',
+          },
+        ];
+
+        this.meshTunnelConf = [
+          {
+            fileExtensions: '.ovpn,.cnf,text/*',
+            property: 'config',
+            tunnelType: 'meshTunnel',
+            required: true,
+          },
+          {
+            fileExtensions: '.crt,text/*',
+            tunnelType: 'meshTunnel',
+            property: 'cert',
+          },
+          {
+            fileExtensions: '.crt,text/*',
+            tunnelType: 'meshTunnel',
+            property: 'cacert',
+          },
+          {
+            fileExtensions: '.key,text/*',
+            tunnelType: 'meshTunnel',
+            property: 'key',
+          },
+          {
+            fileExtensions: '.key,text/*',
+            tunnelType: 'meshTunnel',
+            property: 'takey',
+          },
+        ];
 
         $scope.$watch('$ctrl.internet', this.updateFromInput.bind(this), true);
         $scope.$watch('$ctrl.newInternet', this.updateOutput.bind(this), true);
       }
 
+      toggleVpnList() {
+        this.showVpnList = !this.showVpnList;
+      }
+
+      setTunnelType(type) {
+        if (!this.newInternet.internetTunnel) {
+          this.newInternet.internetTunnel = {};
+        }
+        this.newInternet.internetTunnel.type = type;
+      }
+
+      setMeshTunnelType(type) {
+        if (!this.newInternet.meshTunnel) {
+          this.newInternet.meshTunnel = {};
```

Comment:
*Share internet* is not about having internet access but about sharing your internet connection. Even if you don't want to share it your router may be connected to the internet and you may want to connect with other mesh nodes via a tunnel, right?